src/is_wire/core/message.py

```diff
@@ -0,0 +1,363 @@
+        pretty = "{\n"
+        pretty += "  topic = '{}'\n".format(self.topic or "")
+        pretty += "  created_at = {}\n".format(created_at)
+        pretty += "  correlation_id = {}\n".format(self.correlation_id)
+        pretty += "  reply_to = '{}'\n".format(self.reply_to or "")
+        pretty += "  subscription_id = '{}'\n".format(self.subscription_id
+                                                      or "")
+        pretty += "  timeout = {}\n".format(self.timeout)
+        pretty += "  status = {}\n".format(self.status)
+        pretty += "  metadata = {}\n".format(self.metadata)
+        pretty += "  content_type = {}\n".format(self.content_type)
+        pretty += "  body[{}] = {} \n".format(len(self.body), repr(self.body))
+        pretty += '}\"'
+        return pretty
+
+    def short_string(self):
+        """ Converts a message to a simplified string of its properties, empty
+         fields are not printed """
+        created_at = datetime.fromtimestamp(self.created_at)
+        pretty = "{"
+        pretty += "topic='{}'".format(self.topic)
+        pretty += " created_at={}".format(created_at)
+        if self.has_correlation_id():
+            pretty += " correlation_id={}".format(self.correlation_id)
+        if self.has_reply_to():
+            pretty += " reply_to='{}'".format(self.reply_to)
+        if self.has_subscription_id():
+            pretty += " subscription_id='{}'".format(self.subscription_id)
+        if self.has_timeout():
+            pretty += " timeout={}".format(self.timeout)
+        if self.has_status():
+            pretty += " status={}".format(self.status)
+        if self.has_metadata():
+            pretty += " metadata={}".format(self.metadata)
+        if self.has_content_type():
+            pretty += " content_type={}".format(self.content_type)
+        pretty += " body[{}]={}".format(len(self.body), repr(self.body))
+        pretty += "}"
+        return pretty
+
+    def __eq__(self, other):
+        """ Returns True if the messages are equal, False otherwise """
+        return self.__dict__ == other.__dict__
+
+    def create_reply(self):
+        reply = Message()
+        if self.has_reply_to():
+            reply.topic = self.reply_to
+        if self.has_correlation_id():
+            reply.correlation_id = self.correlation_id
+        if self.has_content_type():
+            reply.content_type = self.content_type
+        return reply
+
+    # topic
+
+    @property
+    def topic(self):
+        """ str: Topic where the message was published or is going
+         to be published """
+        return self._topic
+
+    @topic.setter
+    def topic(self, topic):
+        assert_type(topic, string_types, "topic")
+        self._topic = topic
+
+    def has_topic(self):
+        """ Returns: True if the property topic of the message is set,
+         False otherwise """
+        return self._topic is not None and len(self._topic) != 0
```

Comment:
I had different behaviour in the situations bellow: 
1) Got an exception due to empty topic name
```python
m = Message()
m.body = "abc".encode('latin')
m.topic = ""
c.publish(m)
```
2) Able to publish with an empty topic as well as consume the message with an empty subscription
```python
m = Message()
m.body = "abc".encode('latin')
c.publish(m, topic="")
```
You may raise an exception when publish with an empty topic or remove the length condition on the commented line.